x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,212 @@
+outputs:
+  default:
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+    # Not supported at first
+    queue:
+      type: disk
+
+  long_term_storage:
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+    queue:
+      type: disk
+
+  monitoring:
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+
+settings.monitoring:
+  use_output: monitoring
+
+datasources:
+  # use the nginx package
+  - id?: nginx-x1
+    title?: "This is a nice title for human"
+    # Package this config group is coming from. On importing, we know where it belongs
+    # The package tells the UI which application to link to
+    package?:
+      name: epm/nginx
+      version: 1.7.0
+    namespace?: prod
+    constraints?:
+      # Contraints look are not final
+      - os.platform: { in: "windows" }
+      - agent.version: { ">=": "8.0.0" }
+    use_output: long_term_storage
+    inputs:
```

Comment:
Endpoint also has a requirement for a datasource to be stopped or uninstalled. Maybe we need to add a key for state like "running," "stopped", "uninstalled," etc?

https://github.com/elastic/endpoint-app-team/issues/129

Reply:
when datasource is not needed it should not be part of the config. so endpoint or anything else should be able to detect that it's running something it's not supposed to

Reply:
@michalpristas @mostlyjason I wasn't clear in the configuration I should have added an optional `enabled` key at the datasource level, I have an example for a [stream](https://github.com/elastic/beats/pull/15940/files#diff-043e80bbcc20fded11325ef31433396bR83)

So I the following will happens, _enabled_ default to true, this is indeed common or used behavior for existing beats. If its explicit set to **false**  we disable either the _stream_ or the complete _datasource_. Think about it as a table with a checkbox in the UI.

Concerning uninstalling, as @michalpristas said this is different than a toggle key and should not be present in the configuration.